aries_cloudagent/protocols/connections/v1_0/messages/connection_request.py

```diff
@@ -59,6 +63,16 @@ class Meta:
         description="Label for connection request",
         example="Request to connect with Bob",
     )
+    my_role = fields.Str(
+        required=False,
+        description="My Role that needs to be passed",
+        example="ENDORSER",
```

Comment:
Redundant: there are exactly two roles: inviter and invitee (RFC 23) equivalent to responder and requester (RFC 160). If their_role is one then the other is their_role.flip().

Maybe these refer to tx_roles?

Reply:
I understand that the RFC only has two roles inviter and invitee and transaction roles are something with is neither Indy specific nor in RFC, but they are related to the transactions that are going to be performed between these parties. Do you suggest creating a different record type that handles these roles?

Reply:
Yes - you can attach them as per ConnRecord.attach_invitation()`, `ConnRecord.retrieve_invitation()`. I think it is a mistake to throw this distinct content into the basic connection record.

@ianco what do you think?

Reply:
Yes I agree adding as an attachment makes sense.

Reply:
As attachments are good. We'll update it